Remove "API_KEY" set from ENV Variable in Capistrano and airbrake:deploy

midN opened the pull request:
Removed "API_KEY" param being set in Capistrano tasks and airbrake:deploy task
due to possible conflict with ENV and since it's not needed there and is not
mentioned in Docs regarding deployment.

**Bug / Conflict with ENV variable description:**
If you have ENV variable called API_KEY either locally during deployment or
in your application used somewhere else, Airbrake.configuration.api_key will
be ignored and ENV variable called API_KEY used instead.

mmcdaris commented:
Hi @midN Thanks for your contribution to the Airbrake gem!
I agree that we should remove `API_KEY` since the `Airbrake.configuration.api_key` is already going to give the best value.

I don't see a good reason to prefer `API_KEY` which may have nothing to do with airbrake and is not documented in wiki doc for [tracking deploys](https://github.com/airbrake/airbrake/wiki/Tracking-deployments-with-Airbrake)

overwriting `Airbrake.configuration.api_key` with whatever value is in env var `API_KEY` would just cause unexpected deploy notification failures that would not be very much fun to troubleshoot.

@shifi do you agree this is safe to remove?

shifi commented:
@midN @mmcdaris this is a good improvement.
Can you also update [`airbrake_tasks.rb`](https://github.com/airbrake/airbrake/blob/master/lib/airbrake_tasks.rb#L17)?
I don't think we will need this part anymore: `opts.delete(:api_key)`
Thank you.

midN commented:
@mmcdaris @shifi Thank you for the feedback!
Removed api_key from opts in AirbrakeTasks.deploy and also removed 1 test that was checking if passing api_key opt to AirbrakeTasks.deploy will be used for sending notification.

shifi commented:
Awesome! Thanks a lot @midN
